server/src/service-layer/controllers/PaymentController.ts

```diff
@@ -237,4 +243,169 @@ export class PaymentController extends Controller {
+          stripeCustomerId,
+          CheckoutTypeValues.BOOKING
+        )
+        if (activeSession) {
+          this.setStatus(200)
+          return {
+            stripeClientSecret: activeSession.client_secret,
+            message: "Existing booking checkout session found"
+          }
+        }
+      }
+
+      const bookingSlotService = new BookingSlotService()
+      const bookingDataService = new BookingDataService()
+
+      const bookingSlots =
+        await bookingSlotService.getBookingSlotsBetweenDateRange(
+          dateToFirestoreTimeStamp(
+            new Date(firestoreTimestampToDate(startDate).toDateString())
+          ),
+          dateToFirestoreTimeStamp(
+            new Date(firestoreTimestampToDate(endDate).toDateString())
+          )
+        )
+
+      if (bookingSlots.length !== totalDays) {
+        this.setStatus(409)
+        return {
+          error: "No booking slot available for one or more dates."
+        }
+      }
+
+      const baseAvailabilities =
+        await bookingDataService.getAvailabilityForUser(
+          uid,
+          datesInBooking,
+          bookingSlots
+        )
+      if (baseAvailabilities.some((slot) => !slot)) {
+        this.setStatus(409)
+        return {
+          error: "User has already booked a slot or there is no availability"
+        }
+      }
+
+      const MINUTES_AGO = 30
+      // Lets check for open sessions here:
+      const openSessions = await stripeService.getRecentActiveSessions(
+        CheckoutTypeValues.BOOKING,
+        MINUTES_AGO
+      )
+
+      const currentlyInCheckoutSlotIds = openSessions.flatMap((session) =>
+        JSON.parse(session.metadata[BOOKING_SLOTS_KEY])
+      ) as Array<string>
+
+      const slotOccurences = BookingUtils.getSlotOccurences(
+        currentlyInCheckoutSlotIds
+      )
+
+      const outOfStockBecauseSessionActive = baseAvailabilities.some(
+        (availability) =>
+          availability.baseAvailability - slotOccurences[availability.id] <= 0
+      )
+
+      if (outOfStockBecauseSessionActive) {
+        this.setStatus(409)
+        return {
+          error:
+            "Someone may currently have this item in cart, please try again later"
+        }
+      }
+
+      // implement pricing logic
+      const requiredBookingType =
+        BookingUtils.getRequiredPricing(datesInBooking)
+
+      const requiredBookingProducts = await stripeService.getProductByMetadata(
+        LODGE_PRICING_TYPE_KEY,
+        requiredBookingType
+      )
+      const requiredBookingProduct = requiredBookingProducts.find(
+        (product) => product.active
+      )
+      const { default_price } = requiredBookingProduct
+
+      const clientSecret = await stripeService.createCheckoutSession(
+        uid,
+        `${process.env.FRONTEND_URL}/booking/success?session_id={CHECKOUT_SESSION_ID}&startDate=${datesInBooking[0].toISOString().split("T")[0]}&endDate=${datesInBooking[totalDays - 1].toISOString().split("T")[0]}`,
```

Comment:
This might display incorrect dates based on TZ settings, I think? Depends if the client correctly parses this as a UTC date and not a local date.

Reply:
the client parses this as a string, so i think if we get any off by ones can just fix it on here